baremaps-server-ogcapi/src/main/java/com/baremaps/server/ogcapi/CollectionsResource.java

```diff
@@ -139,79 +108,10 @@ public Response updateCollection(UUID collectionId, Collection collection) {
         handle ->
             handle
                 .createUpdate(
-                    "update collections set title = :title, description = :description, links = cast(:links as jsonb[]), extent = cast(:extent as jsonb), item_type = :item_type, crs = :crs where id = :id")
-                .bind("title", collection.getTitle())
-                .bind("description", collection.getDescription())
-                .bindArray("links", Link.class, collection.getLinks())
-                .bindByType("extent", collection.getExtent(), EXTENT)
-                .bind("item_type", collection.getItemType())
-                .bindArray("crs", String.class, collection.getCrs())
+                    "update collections set collection = cast(:collection as jsonb) where id = :id")
                 .bind("id", collectionId)
+                .bindByType("collection", collection, COLLECTION)
```

Comment:
Have you tried the bindBean method? For simple objects it may be an alternative to the RowMapper.

https://jdbi.org/#_binding_arguments

Reply:
Yes, if the Collection properties are only "known" types, this may work.